cmd/kubeadm/app/util/config/cluster_test.go

```diff
@@ -730,6 +730,9 @@ func TestGetInitConfigurationFromCluster(t *testing.T) {
 			if cfg.ClusterConfiguration.KubernetesVersion != k8sVersionString {
 				t.Errorf("invalid ClusterConfiguration.KubernetesVersion")
 			}
+			if cfg.NodeRegistration.ImagePullPolicy != kubeadmapiv1.DefaultImagePullPolicy {
+				t.Errorf("invalid cfg.NodeRegistration.ImagePullPolicy %v", cfg.NodeRegistration.ImagePullPolicy)
+			}
```

Comment:
this unit test:
- ensures defaulting happens
- adds test coverage for the regression / bug

Reply:
UT would be valuable.

Recently, there are several PRs that pass k/k CIs and break kubeadm CI. It's hard to catch this problem during PR review.

Reply:
the unit test here is sufficient to catch the exact problem.

as far as CI goes, the CI here uses kind and it skips preflight (where the pre-pull failures were) and it does not do "kubeadm upgrade", ideally we'd need a kinder based CI.